src/lava/lib/dl/slayer/block/base.py

```diff
@@ -1428,18 +1430,21 @@ def forward(self, x):
             z = self.input_synapse(x)
         else:
             z = self.input_synapse(x) + self.bias
-        x = torch.zeros_like(z).to(x.device)
 
         spike = torch.zeros(z.shape[:-1]).to(x.device)
 
         if z.shape[0] == self.spike_state.shape[0]:
             spike = spike + self.spike_state
 
-        for time in range(z.shape[-1]):
-            dendrite = z[..., time:time + 1]
-            feedback = self.recurrent_synapse(spike.reshape(dendrite.shape))
-            spike = self.neuron(dendrite + feedback)
-            x[..., time:time + 1] = spike
+        ###
```

Comment:
Generally we wouldn't merge commented code unless it is explanation code that is part of a larger comment.